vllm/config.py

```diff
@@ -429,6 +430,19 @@ def verify_with_parallel_config(
                            "pipeline parallelism currently. Disabling it.")
             self.use_async_output_proc = False
 
+    @cached_property
+    def is_attention_free(self) -> bool:
+        """Returns True if the model has no attention, i.e. the model has no
+        state that grows with the size of the context.
+        """
+
+        # Return true if the model is mamba.
+        # This check should be augmented with more models in the future,
+        # and made more robust if possible.
+        return hasattr(
+            self.hf_text_config,
+            "model_type") and self.hf_text_config.model_type == 'mamba'
```

Comment:
This is fine for now but it may be best to handle in the model registry, kind of like embedding model detection https://github.com/vllm-project/vllm/blob/d5fbb8706d2c7fd00b64cff2efbe7c771fe82c3c/vllm/model_executor/models/registry.py